File: app/src/main/java/dev/eyosiyas/smsblocker/util/Constant.java
package dev.eyosiyas.smsblocker.util;

import android.Manifest;
import android.net.Uri;
import android.provider.ContactsContract;

public class Constant {
    public static final String DATE_PATTERN = "MMM dd yyyy";
    public static final String TIME_PATTERN = "hh:mm:ss a";
    public static final String DATE_TIME_PATTERN = "MMM dd yyyy hh:mm:ss a";
    public static final int PERMISSION_REQUEST_READ_SMS = 2010;
    public static final int PERMISSION_REQUEST_READ_CONTACTS = 2011;
    public static final int REQUEST_SETTING = 2013;
    public static final int REQUEST_SELECT_CONTACT = 2014;
    public static final String READ_CONTACTS = Manifest.permission.READ_CONTACTS;
    public static final String READ_SMS = Manifest.permission.READ_SMS;
    public static final String SEND_SMS = Manifest.permission.SEND_SMS;
    public static final String NUMBER = ContactsContract.CommonDataKinds.Phone.NUMBER;
    public static final String FILTER_SMS_DELIVERED = "SMS_DELIVERED";
    public static final String FILTER_SMS_SENT = "SMS_SENT";
    public static final String FIELD_NAME = "address";
    public static final String FIELD_BODY = "body";
    public static final String FIELD_READ = "read";
    public static final String FIELD_SEEN = "seen";
    public static final String FIELD_DATE = "date";
    public static final String FIELD_TYPE = "type";
    public static final String FIELD_RECEIVED = "received";
    public static final String FIELD_SENT = "sent";
    public static final Uri CONTENT_PROVIDER_SMS = Uri.parse("content://sms/");
    public static final Uri CONTENT_PROVIDER_INBOX = Uri.parse("content://sms/inbox");
    public static final String SMS_BUNDLE = "pdus";
    public static final String NUCLEAR = "nuclear";
    public static final String STARTS_WITH = "startsWith";
    public static final String ENDS_WITH = "endsWith";
}

File: app/src/main/java/dev/eyosiyas/smsblocker/util/Constant.kt
package dev.eyosiyas.smsblocker.util

import android.Manifest
import android.net.Uri
import android.provider.ContactsContract
import android.text.InputFilter

object Constant {
    const val ID: String = "ID"
    const val FIRST_RUN: String = "FirstRun"
    const val LANGUAGE: String = "Language"
    const val BLOCKING_RULE: String = "Blocking Rule"
    const val DATE_PATTERN: String = "MMM dd yyyy"
    const val TIME_PATTERN: String = "hh:mm:ss a"
    const val DATE_TIME_PATTERN: String = "MMM dd yyyy hh:mm:ss a"
    const val STARTS_WITH_TAG: String = "SW"
    const val ENDS_WITH_TAG: String = "EW"
    const val NUCLEAR_OPTION_TAG: String = "NO"
    const val PERMISSION_REQUEST_READ_SMS: Int = 2010
    const val PERMISSION_REQUEST_READ_CONTACTS: Int = 2011
    const val REQUEST_SETTING: Int = 2013
    const val REQUEST_SELECT_CONTACT: Int = 2014
    const val READ_CONTACTS: String = Manifest.permission.READ_CONTACTS
    const val READ_SMS: String = Manifest.permission.READ_SMS
    const val SEND_SMS: String = Manifest.permission.SEND_SMS
    const val NUMBER: String = ContactsContract.CommonDataKinds.Phone.NUMBER
    const val FILTER_SMS_DELIVERED: String = "SMS_DELIVERED"
    const val FILTER_SMS_SENT: String = "SMS_SENT"
    const val FIELD_NAME: String = "address"
    const val FIELD_BODY: String = "body"
    const val FIELD_READ: String = "read"
    const val FIELD_SEEN: String = "seen"
    const val FIELD_DATE: String = "date"
    const val FIELD_TYPE: String = "type"
    const val FIELD_RECEIVED: String = "received"
    const val FIELD_SENT: String = "sent"
    val CONTENT_PROVIDER_SMS: Uri = Uri.parse("content://sms/")
    val CONTENT_PROVIDER_INBOX: Uri = Uri.parse("content://sms/inbox")
    const val SMS_BUNDLE: String = "pdus"
    const val NUCLEAR: String = "nuclear"
    const val STARTS_WITH: String = "startsWith"
    const val ENDS_WITH: String = "endsWith"
    val INPUT_LIMIT = InputFilter.LengthFilter(15)
}